lib/payment_dupes.rb

```diff
@@ -110,7 +143,10 @@ def self.remove_payment_dupes(np_id, designations_to_become_comments)
                                     temp_duplicate_payments = []
                                     temp_offsite_matches = []
                                     recurring_payments.find_each do |recurring_payment|
-                                        equivalent_offsite = s.payments.not_matched.where("kind = 'OffsitePayment' AND gross_amount = ? AND to_char(payments.date, 'YYYY-MM-DD') = ?", recurring_payment.gross_amount, recurring_payment.date.strftime('%Y-%m-%d')).joins(:journal_entries_to_item)
+                                        equivalent_offsite = s.payments.not_matched.where(
+                                            "kind = 'OffsitePayment' AND (gross_amount = ? OR gross_amount = ?) AND (to_char(payments.date, 'YYYY-MM-DD') = ? OR to_char(payments.date, 'YYYY-MM-DD') = ?)",
+                                            recurring_payment.gross_amount, recurring_payment.net_amount, recurring_payment.date.in_time_zone(nonprofit.timezone).strftime('%Y-%m-%d'), recurring_payment.date.strftime('%Y-%m-%d')
+                                        ).joins(:journal_entries_to_item)
```

Comment:
Compares the offsite date with the online date in UTC and in the nonprofit's timezone.

Reply:
I have similar questions as above, I don't understand what the conditions actually are.

Reply:
See this comment https://github.com/CommitChange/houdini/pull/268/files#r810294347